### ISSUE TITLE
Configurable product with swatches, doesn't show tier pricing when clicking edit in cart OR when active swatch set with url hash

### ISSUE THREAD

**Issue by pixiemediaweb** [magento/magento2#28270]
### Preconditions (*)

1. Magento 2.3.5

### Steps to reproduce (*)

1. Create a configurable product with swatch options
2. Add tier prices to one of the options
3. Go to configurable product PDP 
4. Select the tier price product
5. Add the product to cart
6. Click on edit button from mini cart/ cart page
7. Check the tier price is not shown on edit product page
8. Load the page with #attribute_id=value_id to preset the swatch value

### Expected result (*)

1. Tier price block should show

### Actual result (*)

1. Tier price does not show

### Solution (*)

The issue lies in file;
`/vendor/magento/module-swatches/view/base/web/js/swatch-renderer.js`

The function _UpdatePrice: function () { has a dependency on if this.options.tierPriceTemplate being defined on hashed options being set on load, which it isnt.

Here's the change to ensure it gets defined;

Before:

```
_UpdatePrice: function () {
        var $widget = this,
            $product = $widget.element.parents($widget.options.selectorProduct),
            $productPrice = $product.find(this.options.selectorProductPrice),
            result = $widget._getNewPrices(),
            tierPriceHtml,
            isShow;

        $productPrice.trigger(
            'updatePrice',
            {
                'prices': $widget._getPrices(result, $productPrice.priceBox('option').prices)
            }
        );

        isShow = typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount;

        $product.find(this.options.slyOldPriceSelector)[isShow ? 'show' : 'hide']();

        if (typeof result != 'undefined' && result.tierPrices && result.tierPrices.length) {
            if (this.options.tierPriceTemplate) {
                tierPriceHtml = mageTemplate(
                    this.options.tierPriceTemplate,
                    {
                        'tierPrices': result.tierPrices,
                        '$t': $t,
                        'currencyFormat': this.options.jsonConfig.currencyFormat,
                        'priceUtils': priceUtils
                    }
                );
                $(this.options.tierPriceBlockSelector).html(tierPriceHtml).show();
            } 

```
After;

```
_UpdatePrice: function () {
        var $widget = this,
            $product = $widget.element.parents($widget.options.selectorProduct),
            $productPrice = $product.find(this.options.selectorProductPrice),
            result = $widget._getNewPrices(),
            tierPriceHtml,
            isShow;

        $productPrice.trigger(
            'updatePrice',
            {
                'prices': $widget._getPrices(result, $productPrice.priceBox('option').prices)
            }
        );

        isShow = typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount;

        $product.find(this.options.slyOldPriceSelector)[isShow ? 'show' : 'hide']();

        if (typeof result != 'undefined' && result.tierPrices && result.tierPrices.length) {
            // ADDED HERE
            if (this.options.tierPriceTemplate = $(this.options.tierPriceTemplateSelector).html()) {
                tierPriceHtml = mageTemplate(
                    this.options.tierPriceTemplate,
                    {
                        'tierPrices': result.tierPrices,
                        '$t': $t,
                        'currencyFormat': this.options.jsonConfig.currencyFormat,
                        'priceUtils': priceUtils
                    }
                );
                $(this.options.tierPriceBlockSelector).html(tierPriceHtml).show();
            }
```

Additional information
more detailed steps and screens #28270 ([Comment](https://github.com/magento/magento2/issues/28270#issuecomment-648931001))

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28270-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28270-2-4-develop.instances.magento-community.engineering/admin_bac1](https://i-28270-2-4-develop.instances.magento-community.engineering/admin_bac1)
Login: `9a4a9882` Password: `351d853ea175`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by swathishs** [comment]
@pixiemediaweb: I have created a configurable product with color swatch options and tried to pre-set the value. But couldn't able to select the red option in my scenario. Please find the attachment below for reference 
![image](https://user-images.githubusercontent.com/58288685/83419450-b3d37d00-a442-11ea-9172-8a68d07ca963.png)
Link-https://i-28461-2-4-develop.instances.magento-community.engineering/config-product-by-ajith.html#color=red

Could you give some more information on how to pre-set the  #attribute_id=value_id

**Comment by swathishs** [comment]
Hi @pixiemediaweb 
Feel free to reopen once you update the comment 
Thanks !!

**Comment by pixiemediaweb** [comment]
@swathishs - you need to use the attribute ID and value ID in the hash.  It will look something like #123=56 where 123 is the attribute ID for colour and 56 is the value for red

**Comment by pixiemediaweb** [comment]
> Hi @pixiemediaweb
> Feel free to reopen once you update the comment
> Thanks !!

I cant reopen it because you closed it.  I provided the complete cause and the solution in my original post, can you reopen please

**Comment by swathishs** [comment]
@pixiemediaweb : Could you please provide extra steps to reproduce the issue  and it will be helpful if you put the screenshot

**Comment by pixiemediaweb** [comment]
Hi @swathishs - no problem, might be easier if you could resurrect a demo instance with a configurable swatch product so I can show you how M2 natively supports pre-selecting of configurable options by their option and value IDs

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28270-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28270-2-4-develop.instances.magento-community.engineering/admin_de9e](https://i-28270-2-4-develop.instances.magento-community.engineering/admin_de9e)
Login: `27b16d64` Password: `41d56df9d1bf`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
@pixiemediaweb : You can use the above instance.

**Comment by pixiemediaweb** [comment]
Here you go;

Config swatch product: 
https://i-28270-2-4-develop.instances.magento-community.engineering/testconfig.html

Choose the blue swatch - that has qty breaks and they show.

Now try this:
https://i-28270-2-4-develop.instances.magento-community.engineering/testconfig.html#194=280

^^ that's how you preset the chosen swatch.  The price breaks dont show.

Exactly the same problem if you select blue and add to basket.  Now go to the basket and click the edit link. You go back to the product, blue is selected, but prices dont show.

  The fix is in my original post on this thread.

**Comment by pixiemediaweb** [comment]
![image](https://user-images.githubusercontent.com/6330277/85573323-b1bda200-b62d-11ea-9de1-440d76de0540.png)

**Comment by pixiemediaweb** [comment]
should show;
![image](https://user-images.githubusercontent.com/6330277/85573372-bda96400-b62d-11ea-940a-d94fd16e8ce0.png)

**Comment by swathishs** [comment]
@pixiemediaweb : The issue is reproducible in Magento 2.4 develop instance. Can you let me know which scenario ur getting this URL?
I got one scenario after adding that blue product to the cart and click on edit from cart page its redirect to update pdp page. The tier price is not shown.
![image](https://user-images.githubusercontent.com/58288685/85590696-3fb38000-b662-11ea-8a15-f3c2768b0239.png)

**Comment by pixiemediaweb** [comment]
We use it to improve configurable product usability, carrying the selected option in a list to the product page.   But it's functionality that already exists in the native file;

/vendor/magento/module-swatches/view/base/web/js/swatch-renderer.js

Look for function _getSelectedAttributes: function () { .... }

I believe the same functionality is being used when returning to the product from the cart via the edit button

**Comment by swathishs** [comment]
Can we change the title to "Configurable product with swatches, doesn't show tier pricing when we redirect to update product page via edit button from cartpage "

**Comment by swathishs** [comment]
The issue is reproducible in Magento 2.4 develop instance 

Please find the attachment below:

1. Product detail page with tier price product:
![pdp](https://user-images.githubusercontent.com/58288685/85593185-6f638780-b664-11ea-81ec-b3180de03945.png)

2. Mini cart:
![mini](https://user-images.githubusercontent.com/58288685/85593402-a0dc5300-b664-11ea-9124-ea83b82cfd3a.png)

3. Update product detail page:
![update](https://user-images.githubusercontent.com/58288685/85593465-b3568c80-b664-11ea-9837-9b3b5f6921c4.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @swathishs 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-35432` were created

**Issue Available**: @swathishs, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by srsathish92** [comment]
@magento I am working on this

**Comment by engcom-Alfa** [comment]
Hi @pixiemediaweb . Thanks for collaboration. This PR was reviewed on Community Contribution Triage meeting. The recording of the meeting you can look by the [link ](https://www.youtube.com/watch?v=4dFuALJ0L4I&t=265s)

**Comment by slavvka** [comment]
Hi @pixiemediaweb. Thank you for your report. 
The issue has been fixed in magento/magento2#29137 by @srsathish92 in 2.4-develop branch
Related commit(s):
 - [2765164f13afc40279a474a2f255a84ae3d2bf47](https://github.com/magento/magento2/commit/2765164f13afc40279a474a2f255a84ae3d2bf47)
 - [d516e27b844e41066dcb1afdcf26eaa3595fad1d](https://github.com/magento/magento2/commit/d516e27b844e41066dcb1afdcf26eaa3595fad1d)
 - [c5e4aa5acfcdfb37a950c338bae970831ee67ab4](https://github.com/magento/magento2/commit/c5e4aa5acfcdfb37a950c338bae970831ee67ab4)
 - [13f5caebd21773d2b476096f82dda7abc9dcfe5e](https://github.com/magento/magento2/commit/13f5caebd21773d2b476096f82dda7abc9dcfe5e)
 - [59c187f03817f5c318a5829e67974143ea99abbc](https://github.com/magento/magento2/commit/59c187f03817f5c318a5829e67974143ea99abbc)
 - [7d0d91d54f65e9e13b876588f4a606f6eb92b7f8](https://github.com/magento/magento2/commit/7d0d91d54f65e9e13b876588f4a606f6eb92b7f8)
 - [22c74f99a4a746e3545ae3dc7ae22780e883a474](https://github.com/magento/magento2/commit/22c74f99a4a746e3545ae3dc7ae22780e883a474)
 - [441e2cdba53c7fb42a0ef0063294edfa7af9b6f2](https://github.com/magento/magento2/commit/441e2cdba53c7fb42a0ef0063294edfa7af9b6f2)

The fix will be available with the upcoming 2.4.1 release.
